src/test/java/com/google/sps/email/EmailTesting.java

```diff
@@ -0,0 +1,173 @@
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.jvnet.mock_javamail.Mailbox;
+
+import java.util.List;
+
+import com.google.sps.testing.EmailServlet;
+import com.google.sps.testing.GoodDeed;
+
+@RunWith(JUnit4.class)
+public class EmailTesting {
+    private EmailServlet emailServlet;
+
+    private static final String USER = "User";
+    private static final String EMAIL = "email";
+    private static final String EMAIL_INPUT = "test@gmail.com";
+    private static final String EMAIL_SUBJECT = "Complete your daily deed";
+
+    private static final String GOOD_DEED = "GoodDeed";
+    private static final String NAME = "Name";
+    private static final String DESCRIPTION = "Description";
+    private static final String POSTED_YET = "Posted Yet";
+    private static final String DAILY_DEED = "Daily Deed";
+    private static final String TIMESTAMP = "Timestamp";
+    private static final String LINK = "Link";
+
+    private static final String TITLE = "Deed Name";
+    private static final String DESCRIPTION_INPUT = "Deed Description";
+    private static final String TRUE_STRING = "true";
+    private static final String FALSE_STRING = "false";
+    private static final long TIMESTAMP_INPUT = 67890;
+    private static final boolean TRUE = true;
+    private static final boolean FALSE = false;
+
+    private final LocalServiceTestHelper helper = 
+        new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+    @Before
+    public void setUp() {
+        emailServlet = new EmailServlet();
+        helper.setUp();
+        MockitoAnnotations.initMocks(this);
+        Mailbox.clearAll();
+    }
+
+    @Test
+    public void testFetchEmails() {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+
+        Entity testEntity = new Entity(USER);
+        testEntity.setProperty(EMAIL, EMAIL_INPUT);
+        ds.put(testEntity);
+
+        Query query = new Query(USER);
+
+        Assert.assertEquals(1, ds.prepare(query).countEntities(withLimit(10)));
+
+        List<String> emails = emailServlet.fetchEmails();
+        String email = emails.get(0);
+
+        Assert.assertTrue(email.equals(EMAIL_INPUT));
+    }
+
+    @Test
+    public void testsendEmail() throws MessagingException, IOException, MessagingException {
```

Comment:
`MessagingException` can removed here since we do catch it in the actual method. Generally, here, you want to define exceptions that maybe thrown but not explicitly caught (also, `MessagingException` appears twice here)